### PR TITLE
mod: upgrade istio deps

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Background
Update the dependencies of our external-apis repository

# Context

## How did I choose these versions?
I used the [gloo-mesh-enterprise go.mod](https://github.com/solo-io/gloo-mesh-enterprise/blob/main/go.mod) as the source of truth, and fast-forwarded the dependencies in this project to match

## How does this impact Gloo Edge?
Gloo Edge Enterprise relies on a [pinned version of external-apis](https://github.com/solo-io/solo-projects/blob/main/go.mod#L428) which is not built off of the main branch.

We have some ongoing work to import gme into solo-projects. To do this correctly, we get the opportunity to align the dependencies in the project.

## How is this tested in Gloo Edge?
https://github.com/solo-io/solo-projects/pull/5742: We can successfully remove the pinned version, and reconcile all imports.

## How is this tested in Gloo Mesh Enterprise?
https://github.com/solo-io/gloo-mesh-enterprise/pull/14631

## What version will we release this in?
There aren't technically breaking changes in this change set, so I'm inclined to release this as a patch release: `v0.2.6`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,75 +1,76 @@
...
-	github.com/cilium/cilium v1.11.5
+	github.com/cilium/cilium v1.15.0-pre.2
 	github.com/golang/mock v1.6.0
 	github.com/pkg/errors v0.9.1
 	github.com/rotisserie/eris v0.1.1
 	github.com/servicemeshinterface/smi-sdk-go v0.4.1
-	github.com/solo-io/skv2 v0.30.0
-	istio.io/client-go v1.17.1
-	istio.io/istio v0.0.0-20221208171804-952b2e8bc4b0
+	github.com/solo-io/skv2 v0.36.3
```
Comment (npolshakova) [inline go.mod:12]: Let's check that this skv2 bump does break with enterprise. It shouldn't, but it's currently using 0.36.1: https://github.com/solo-io/gloo-mesh-enterprise/blob/main/go.mod#L95

File: go.mod [hunk 2]
```diff
@@ -1,75 +1,76 @@
 module github.com/solo-io/external-apis
 
-go 1.21
+go 1.21.5
```
Comment (npolshakova) [inline go.mod:3]: Will adding the patch version here effect the edge side? I know gloo-mesh-enterprise is using patches and golang will do patch version upgrades automatically with the new version manager shipped in 1.21.1, but I thought edge is still using 1.21 in the go.mod

Comment (sam-heilbron) [reply go.mod:3]: Edge Enterprise relies on the same patch version as GME: https://github.com/solo-io/solo-projects/blob/main/go.mod#L3 (1.21.5), so I think we're safe

File: go.mod (hunk 1 above)
Comment (sam-heilbron) [reply go.mod:12]: Good call, I'll open a PR in GME to import this, and prove that it works via CI

Comment (sam-heilbron) [reply go.mod:12]: I updated the PR body to include this detail, but I opened https://github.com/solo-io/gloo-mesh-enterprise/pull/14631 to demonstrate tthis can be pulled in. I'll try to get eyes on that PR

Comment (sam-heilbron) [reply go.mod:12]: CI is passing on that linked PR, so I would consider this check completd, but happy to perform more validation if neeeded

**Review by npolshakova — APPROVED** [review]

**Pull request merged**